pkg/kapp/resources/resource_types.go

```diff
@@ -265,3 +265,16 @@ func NonMatching(in []ResourceType, ref ResourceRef) []ResourceType {
 	}
 	return out
 }
+
+// TODO: Extend ResourceRef and PartialResourceRefd to allow GVK matching
+func MatchingAnyGVK(in []ResourceType, gvks []schema.GroupVersionKind) []ResourceType {
+	var out []ResourceType
+	for _, item := range in {
+		for _, gvk := range gvks {
+			if item.APIResource.Group == gvk.Group && item.APIResource.Version == item.APIResource.Version && item.Kind == gvk.Kind {
```

Comment:
nit: Make it like `item.Kind == gvk.Kind && item.APIResource.Version == item.APIResource.Version && item.APIResource.Group == gvk.Group`. This will be faster than above one.

Reply:
Yep! that makes sense. As of now I plan on extending the functionality [here](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/resources/refs.go) to match Kinds as well. Will keep it in mind then.